heuristic/heuristic_calculations.py

```diff
@@ -1,322 +1,211 @@
-                (
-                    sum(
-                        sum(x[e, t, v] for t, v in shifts_per_day[i_marked])
-                        for i_marked in range(i, i + data["limit_on_consecutive_days"])
-                    )
-                )
-                - data["limit_on_consecutive_days"],
-            )
+    for e in model.employees:
+        for i in range(len(model.days)-model.L_C_D):
+            consecutive_days[e,i] = max(0,(sum(
+                sum(x[e,t,v] for t,v in model.shifts_at_day[i_marked]) 
+            for i_marked in range(i,i+model.L_C_D)))- model.L_C_D)
+            
+            # if(consecutive_days[e,i] != model.q_con[e,i].x):
+            #     print("Different consecutive days")
     return consecutive_days
 
-
-def calculate_f(data, soft_vars, w, employees=None):
-
-    if not employees:
-        employees = data["staff"]["employees"]
-
+def calculate_f(model, soft_vars, w, employees=None):
+    if(employees == None):
+        employees = model.employees
     f = {}
-
-    # todo: add number_of_days to save calculations?
-    days = data["time"]["days"]
-
     for e in employees:
-        f[e] = (
-            sum(w[e, j][1] for j in data["time"]["weeks"])
-            - sum(soft_vars["deviation_contracted_hours"][e, j] for j in data["time"]["weeks"])
-            - sum(soft_vars["partial_weekends"][e, i] for i in data["time"]["saturdays"])
-            - sum(
-                soft_vars["isolated_working_days"][e, i + 1]
-                + soft_vars["isolated_off_days"][e, i + 1]
-                for i in range(len(days) - 2)
-            )
-            - sum(
-                soft_vars["consecutive_days"][e, i]
-                for i in range(len(days) - data["limit_on_consecutive_days"])
-            )
-        )
+        f[e] = (sum(w[e,j][1] for j in model.weeks)
+            - sum(soft_vars["contracted_hours"][e,j] for j in model.weeks)
+            - sum(soft_vars["partial_weekends"][e,i] for i in model.saturdays)
+            - sum(soft_vars["isolated_working_days"][e,i+1] + soft_vars["isolated_off_days"][e,i+1] for i in range(len(model.days)-2))
+            - sum(soft_vars["consecutive_days"][e,i] for i in range(len(model.days)-model.L_C_D)))
     return f
 
 
 def calculate_objective_function(model, soft_vars, w):
-
     f = calculate_f(model, soft_vars, w)
     g = min(f.values())
+    objective_function_value = (sum(f.values()) + g - abs(sum(soft_vars["deviation_from_ideal_demand"].values())))
+    return objective_function_value, f
+
+
+
+
+
+
 
-    objective_function_value = (
-        sum(f.values()) + g - abs(sum(soft_vars["deviation_from_ideal_demand"].values()))
-    )
 
-    return objective_function_value, f
 
 
-# todo: I am regarding all of the following as out of use. -Even
-# Not needed at the moment and are not in use. Might be deleted at a later time when I know for sure.
+#Not needed at the moment and are not in use. Might be deleted at a later time when I know for sure.
```

Comment:
Can you just delete all the following functions? If they are not in use I find them not to contribute anything. If they are needed later we always have Git as an archive ;)

Reply:
We could. This merge have been a bit less structured than I like since there were som many merges. I can clean up the code and remove it before merge